scripts/analytics/node_summary.py

```diff
@@ -0,0 +1,132 @@
+from dateutil.parser import parse
+from datetime import timedelta, datetime
+
+from website.app import init_app
+from website.models import Node
+from website.settings import KEEN as keen_settings
+from keen.client import KeenClient
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def get_events(date):
+    """Count how many nodes exist.
+
+    If the date is today, include all nodes up until the point when the script was called.
+    Also include counts for public and private projects, and embargoed and withdrawn registrations
+    with a date provided. These numbers are only accurate for the current time, since we can't know
+    the public/private status on a given date in the past.
+
+    If a date is given that isn't right now, include all nodes that were created up through the end of that date.
+    Do not include counts for public and private projects, or embargoed and withdrawn registrations.
+    """
+    right_now = datetime.utcnow()
+    is_today = True if right_now.day == date.day and right_now.month == date.month else False
+    if not is_today:
+        date = date + timedelta(1)
+
+    logger.info('Gathering a count of nodes up until {}'.format(date.isoformat()))
+
+    node_query = (
+        Q('is_deleted', 'ne', True) &
+        Q('is_folder', 'ne', True) &
+        Q('date_created', 'lt', date)
+    )
+
+    registration_query = node_query & Q('is_registration', 'eq', True)
+    non_registration_query = node_query & Q('is_registration', 'eq', False)
+    project_query = non_registration_query & Q('parent_node', 'eq', None)
+    registered_project_query = registration_query & Q('parent_node', 'eq', None)
+    public_query = Q('is_public', 'eq', True)
+    private_query = Q('is_public', 'eq', False)
+    retracted_query = Q('retraction', 'ne', None)
+    project_public_query = project_query & public_query
+    project_private_query = project_query & private_query
+    node_public_query = non_registration_query & public_query
+    node_private_query = non_registration_query & private_query
+    registered_node_public_query = registration_query & public_query
+    registered_node_private_query = registration_query & private_query
+    registered_node_retracted_query = registration_query & retracted_query
+    registered_project_public_query = registered_project_query & public_query
+    registered_project_private_query = registered_project_query & private_query
+    registered_project_retracted_query = registered_project_query & retracted_query
+
+    totals = {
+        'nodes': {
+            'total': Node.find(node_query).count(),
+        },
+        'projects': {
+            'total': Node.find(project_query).count(),
+        },
+        'registered_nodes': {
+            'total': Node.find(registration_query).count(),
+        },
+        'registered_projects': {
+            'total': Node.find(registered_project_query).count(),
+        }
+    }
+
+    if is_today:
```

Comment:
It looks like this is building summaries for dates that are not today, and snapshots for dates that are today.  Can we split these out into two scripts or methods and store them in separate collections?